### PR TITLE
Add a Swift sample

### PR DESCRIPTION
**Pull request by SSheldon**
This provides a simple example of interacting with the Yelp API via Swift, using the YelpAPI pod.

I ended up adding an additional dependency and some extensions to make YelpAPI work with futures. I wasn't intending to do this originally, but writing the sample without futures was miserable. So many dispatch groups... *shudders*

### CODE REVIEW

File: fusion/swift/.gitignore [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Build generated
```
Comment (mittonk) [inline fusion/swift/.gitignore]: This example deserves mention in the root `README.md`.

@tomelm or @d9chen , were one of you going to add that yourself or should @SSheldon do it in this pull request?

File: fusion/swift/YelpAPISample/main.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+//  main.swift
+//  YelpAPISample
+//
+
+import YelpAPI
+import BrightFutures
+
+// Fill in your app keys here from
+// https://www.yelp.com/developers/v3/manage_app
+let appId = ""
```
Comment (mittonk) [inline fusion/swift/YelpAPISample/main.swift]: `appId` looks great.

Should we do `appSecret` to match https://www.yelp.com/developers/v3/manage_app and add some namespacing?

File: fusion/swift/YelpAPISample/main.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+// Search for 3 dinner restaurants
+let query = YLPQuery(location: "San Francisco, CA")
+query.term = "dinner"
+query.limit = 3
+
+YLPClient.authorize(withAppId: appId, secret: secret).flatMap { client in
+  client.search(withQuery: query)
+}.onSuccess { search in
+  if let topBusiness = search.businesses.first {
+    print("Top business: \(topBusiness.name), id: \(topBusiness.identifier)")
```
Comment (mittonk) [inline fusion/swift/YelpAPISample/main.swift]: Great, I like this level of functionality for a code sample.

File: fusion/swift/YelpAPISample/main.swift (hunk 2 above)
Comment (SSheldon) [inline fusion/swift/YelpAPISample/main.swift]: Good call, renamed!

File: fusion/swift/.gitignore (hunk 1 above)
Comment (SSheldon) [inline fusion/swift/.gitignore]: No worries, I've added a link in the README

**Comment by mittonk** [comment]
:shipit:

**Comment by jagadeesh583** [comment]
import Foundation

let headers = [
  "authorization": "Bearer youkey",
  "cache-control": "no-cache"
]

let request = NSMutableURLRequest(url: NSURL(string: "https://api.yelp.com/v3/businesses/north-india-restaurant-san-francisco")! as URL,
                                        cachePolicy: .useProtocolCachePolicy,
                                    timeoutInterval: 10.0)
request.httpMethod = "GET"
request.allHTTPHeaderFields = headers

let session = URLSession.shared
let dataTask = session.dataTask(with: request as URLRequest, completionHandler: { (data, response, error) -> Void in
  if (error != nil) {
    print(error)
  } else {
    let httpResponse = response as? HTTPURLResponse
    print(httpResponse)
  }
})

dataTask.resume()

**Pull request merged**